### PR TITLE
mlops-engines: add LocalAI

### PR DESCRIPTION
**Pull request by mudler**
Hi!  :wave: 

Awesome project! I really like the idea of having a single point where there is listed the state of art and what are the most widely used tool in the OSS ecosystem!

I've added [LocalAI](https://github.com/go-skynet/LocalAI) to the list, the free OSS OpenAI alternative.
I hope you will find it useful!

### CODE REVIEW

File: mlops-engines.md [hunk 1]
```diff
@@ -127,6 +128,18 @@ Cons:
...
+## LocalAI
+
+[LocalAI](https://localai.io) is the free, Open Source OpenAI alternative. LocalAI act as a drop-in replacement REST API that’s compatible with OpenAI API specifications for local inferencing. It allows you to run LLMs with various backend such as llama.cpp or vllm, generate images, generate audio, transcribe audio, self-hosted, on-prem with consumer grade hardware, supporting multiple model families. 
+
+Pros:
+- wide range of models supported
+- supports for LocalAI functions (OpenAI functions, but self-hosted)
+- easy to integrate
+
+Cons:
```
Comment (mudler) [inline mlops-engines.md:140]: to reviewers..  I've added a list that I think it's reasonable, but eager to hear more feedback!

**Review by filopedraz — APPROVED** [review]

File: mlops-engines.md [hunk 2]
```diff
@@ -127,6 +128,20 @@ Cons:
...
 This is not an exhaustive list of MLOps engines by any means. There are many other tools and frameworks developer use to deploy their ML models. There is ongoing development in both the open-source and private sectors to improve the performance of LLMs. It's up to the community to test out different services to see which one works best for their use case.
 
+## LocalAI
+
+[LocalAI](https://localai.io) from https://github.com/go-skynet/LocalAI ([not to be confused](https://github.com/louisgv/local.ai/discussions/71) with [](desktop-apps.md#localai) from https://github.com/louisgv/local.ai) is the free, Open Source alternative to OpenAI. LocalAI act as a drop-in replacement REST API that's compatible with OpenAI API specifications for local inferencing. It can run LLMs (with various backend such as https://github.com/ggerganov/llama.cpp or [](#vllm)), generate images, generate audio, transcribe audio, and can be self-hosted (on-prem) with consumer-grade hardware.
+
+Pros:
+
+- wide range of models supported
+- support for "LocalAI functions" (self-hosted verison of OpenAI functions)
```
Comment (casperdcl) [inline mlops-engines.md:138]: Are you sure? based on https://localai.io/features/openai-functions it seems identical, no?

Also missing some links?

```suggestion
- support for [functions](https://localai.io/features/openai-functions) (self-hosted [OpenAI functions](https://platform.openai.com/docs/guides/gpt/function-calling))
```

File: mlops-engines.md [hunk 3]
```diff
@@ -127,6 +128,20 @@ Cons:
 
 This is not an exhaustive list of MLOps engines by any means. There are many other tools and frameworks developer use to deploy their ML models. There is ongoing development in both the open-source and private sectors to improve the performance of LLMs. It's up to the community to test out different services to see which one works best for their use case.
 
+## LocalAI
+
+[LocalAI](https://localai.io) from https://github.com/go-skynet/LocalAI ([not to be confused](https://github.com/louisgv/local.ai/discussions/71) with [](desktop-apps.md#localai) from https://github.com/louisgv/local.ai) is the free, Open Source alternative to OpenAI. LocalAI act as a drop-in replacement REST API that's compatible with OpenAI API specifications for local inferencing. It can run LLMs (with various backend such as https://github.com/ggerganov/llama.cpp or [](#vllm)), generate images, generate audio, transcribe audio, and can be self-hosted (on-prem) with consumer-grade hardware.
+
+Pros:
+
+- wide range of models supported
```
Comment (casperdcl) [inline mlops-engines.md:137]:
```suggestion
- [wide range of models supported](https://localai.io/model-compatibility)
```

File: mlops-engines.md [hunk 4]
```diff
@@ -127,6 +128,20 @@ Cons:
...
+- support for "LocalAI functions" (self-hosted verison of OpenAI functions)
+- easy to integrate
```
Comment (casperdcl) [inline mlops-engines.md:139]: probably needs some explanation

File: mlops-engines.md [hunk 5]
```diff
@@ -127,6 +128,20 @@ Cons:
...
+
+Cons:
+
+- binary version is harder to run and compile locally
```
Comment (casperdcl) [inline mlops-engines.md:143]: link? e.g. some GH issue

**Review by casperdcl — CHANGES_REQUESTED** [review]
thanks! pushed a few edits, please do review @mudler :)

File: mlops-engines.md (hunk 2 above)
Comment (mudler) [reply mlops-engines.md]: Right! I've put that into quotes as it is a LocalAI feature, but good point!

File: mlops-engines.md (hunk 3 above)
Comment (mudler) [reply mlops-engines.md]: 👍

**Pull request merged**